exporter/exporter.go

```diff
@@ -0,0 +1,531 @@
+package exporter
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+	"log"
+	"strconv"
+	"upyun-test/httpRequest"
+)
+
+const cdnNameSpace = "upYun"
+
+
+type cdnExporter struct {
+	domainList              *[]string
+	token                   string
+	rangeTime               int64
+	delayTime               int64
+	cdnRequestCount         *prometheus.Desc
+	cdnBandWidth            *prometheus.Desc
+	cdn4xxErrorRate         *prometheus.Desc
+	cdn5xxErrorRate         *prometheus.Desc
+	cdnResourceBandWidth    *prometheus.Desc
+	cdnResource4xxErrorRate *prometheus.Desc
+	cdnResource5xxErrorRate *prometheus.Desc
+	cdnResource4xxCount     *prometheus.Desc
+	cdnResource5xxCount     *prometheus.Desc
+	cdnResource2xxCount     *prometheus.Desc
+	cdnResource3xxCount     *prometheus.Desc
+	cdnResource200Count     *prometheus.Desc
+	cdnResource206Count     *prometheus.Desc
+	cdnResource301Count     *prometheus.Desc
+	cdnResource303Count     *prometheus.Desc
+	cdnResource304Count     *prometheus.Desc
+	cdnResource400Count     *prometheus.Desc
+	cdnResource403Count     *prometheus.Desc
+	cdnResource404Count     *prometheus.Desc
+	cdnResource411Count     *prometheus.Desc
+	cdnResource499Count     *prometheus.Desc
+	cdnResource500Count     *prometheus.Desc
+	cdnResource502Count     *prometheus.Desc
+	cdnResource503Count     *prometheus.Desc
+	cdnResource504Count     *prometheus.Desc
+}
+
+func CdnCloudExporter(domainList *[]string, token string, rangeTime int64, delayTime int64) *cdnExporter {
+	return &cdnExporter{
+		domainList: domainList,
+		token:      token,
+		rangeTime:  rangeTime,
+		delayTime:  delayTime,
+		cdnRequestCount: prometheus.NewDesc(
+			prometheus.BuildFQName(cdnNameSpace, "cdn", "request_count"),
+			"cdn总请求数",
+			[]string{
+				"instanceId",
+			},
+			nil,
+		),
+		cdnBandWidth: prometheus.NewDesc(
+			prometheus.BuildFQName(cdnNameSpace, "cdn", "bandWidth"),
+			"cdn总带宽(Bps)",
+			[]string{
+				"instanceId",
+			},
+			nil,
+		),
+		cdn4xxErrorRate: prometheus.NewDesc(
+			prometheus.BuildFQName(cdnNameSpace, "cdn", "4xx_error_rate"),
+			"cdn4xx错误率",
+			[]string{
+				"instanceId",
+			},
+			nil,
+		),
+		cdn5xxErrorRate: prometheus.NewDesc(
+			prometheus.BuildFQName(cdnNameSpace, "cdn", "5xx_error_rate"),
+			"cdn5xx错误率",
+			[]string{
+				"instanceId",
+			},
+			nil,
+		),
+		cdnResourceBandWidth: prometheus.NewDesc(
+			prometheus.BuildFQName(cdnNameSpace, "cdn", "resource_bandWidth"),
+			"回源带宽(Bps)",
+			[]string{
+				"instanceId",
+			},
+			nil,
+		),
+		cdnResource4xxErrorRate: prometheus.NewDesc(
+			prometheus.BuildFQName(cdnNameSpace, "cdn", "resource_4xx_error_rate"),
```

Comment:
回源相关的子系统都设置成 "backsource"
```suggestion
			prometheus.BuildFQName(cdnNameSpace, "backsource", "resource_4xx_error_rate"),
```